### PR TITLE
[SIP-4] replace explorer ajax calls with `SupersetClient`

### PR DESCRIPTION
**Pull request by williaster**
This PR is one of a few that implements the final step 4) discussed in #5772 and refactors just **explorer**-specific ajax calls (not including charts) for easier review.

Note that the new `@superset-ui/core` dep + setup for `SupersetClient` is duplicated across all ajax PRs 
- dashboard #5854 
- **explore** (this PR)
- charts #5875
- sql lab #5896
- logging + misc

@kristw @mistercrunch @graceguo-supercat @michellethomas @conglei

### CODE REVIEW

File: superset/assets/spec/helpers/setupSupersetClient.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import fetchMock from 'fetch-mock';
+import { SupersetClient } from '@superset-ui/core';
+
+export default function setupSupersetClient() {
+  // The following are needed to mock out SupersetClient requests
+  // including CSRF authentication and initialization
+  // global.FormData = window.FormData;
```
Comment (williaster) [inline superset/assets/spec/helpers/setupSupersetClient.js]: going to remove this file and add it as a cherry-pick to share across these ajax refactors. this line is also needed.

File: superset/assets/src/explore/components/AdhocFilterEditPopoverSimpleTabContent.jsx [hunk 2]
```diff
@@ -1,7 +1,9 @@
+/* global AbortController */
```
Comment (kristw) [inline superset/assets/src/explore/components/AdhocFilterEditPopoverSimpleTabContent.jsx]: can this be added at project level?

**Review by kristw — APPROVED** [review]
LGTM

**Review by kristw — APPROVED** [review]

**Review by graceguo-supercat — APPROVED** [review]
LGTM

**Pull request merged**